aaelf64-morello/aaelf64-morello.rst

```diff
@@ -866,3 +934,202 @@ veneer is used. The BX changes the execution state from A64 to C64:
+offset and the second 8 bytes the size). This GOT slot is initialized by a
+``R_MORELLO_TPREL128`` dynamic relocation. The access must use the
+``R_MORELLO_TLSIE_ADR_GOTTPREL_PAGE20`` and ``R_MORELLO_TLSIE_ADD_LO12``
+relocations in order to allow relaxation to Local Exec. There are no other
+requirements on how this is performed or the registers used. A possible
+instruction sequence could be:
+
+.. code-block:: text
+
+  adrp c0, :gottprel:initial_exec_var
+  add c0, c0, :gottprel_lo12:initial_exec_var
+  ldp x0, x8, [c0]
+  mrs c1, CTPIDR_EL0
+  add c0, c1, x0, uxtx
+  scbnds c0, c0, x8
+
+Initial Exec to Local Exec relaxation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The linker will generate 16 bytes in a read-only section, containing
+the offset in the static TLS block in the first 8 bytes and the size of the
+symbol in the next 8 bytes:
+
+.. code-block:: text
+
+  .section .rodata
+  _sym_data:
+    .xword tlsoffset(sym)
+    .xword sizeof(sym)
+
+.. note::
+
+  ``tlsoffset(sym)`` denotes the offset in the static TLS block of the symbol
+  ``sym``, while sizeof(sym) denotes the size of the symbol ``sym``. These are
+  not valid assebler directives.
+
+The relaxation is performed by:
+
+- changing the ``R_MORELLO_TLSIE_ADR_GOTTPREL_PAGE20`` relocation on the symbol
+  ``sym`` to a ``R_MORELLO_ADR_PREL_PG_HI20`` with the symbol ``_sym_data``
+
+- changing the ``R_MORELLO_TLSIE_ADD_LO12`` relocation on symbol the ``sym`` to
+  a ``R_AARCH64_ADD_ABS_LO12_NC`` relocation with the symbol ``_sym_data``.
+
+.. note::
+
+  The symbol and section names in the example above are only used for explanation
+  purposes. An implementation does not need to create an additional symbol when
+  performing this relaxation. There is no constraint on the name of the read-only
+  section where the data is placed.
+
+General Dynamic
+^^^^^^^^^^^^^^^
+
+The instruction sequence used for the General Dynamic access model is similar to
+that of other TLSDESC implementations, with the exception that the result
+doesn't need to be added to the thread pointer. However c2 needs to contain
+the thread pointer. The instruction sequence contains an additional NOP
+instruction in order to permit the static linker to perform a relaxation to Local
+Exec or Initial Exec.
+
+The General Dynamic access sequence must be output in the following form to
+allow correct linker relaxation:
+
+.. code-block:: text
+
+  adrp c0, :tlsdesc:sym
+  ldr c1, [c0, :tlsdesc_lo12:sym]
+  add c0, c0, :tlsdesc_lo12:sym
+  nop
+  .tlsdesccall sym
+  blr c1
+
+General Dynamic to Initial Exec relaxation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The relaxed sequence is:
+
+.. code-block:: text
+
+  adrp c0, :gottprel:sym
+  add c0, c0, :gottprel_lo12:sym
+  ldp x0, x1, [c0]
+  add c0, c2, x0
+  scbnds c0, c0, x1
+
+
+General Dynamic to Local Exec relaxation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The linker will generate 16 bytes in the ``.rodata`` section, containing
```

Comment:
Just since we've adjusted the earlier phrase from "the `.rodata` section" to "a read-only section" it would be nice to apply the same change here.